### PR TITLE
[tests-only] add test download previos version

### PR DESCRIPTION
**Pull request by ScharfViktor**
A test for existing error [2261](https://github.com/owncloud/ocis/issues/2261) is written
And added to expected-failures

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles1</strong> https://drone.owncloud.com/owncloud/web/20384/56/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/20386/17/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:15
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingInternalUsers1</strong> https://drone.owncloud.com/owncloud/web/20393/59/1
The following scenarios passed on retry:
- webUISharingInternalUsers/shareWithUsers.feature:158

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/20393/16/1
The following scenarios passed on retry:
- webUISharingAcceptSharesToRoot/acceptShares.feature:108

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles1</strong> https://drone.owncloud.com/owncloud/web/20393/56/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: tests/acceptance/expected-failures-with-ocis-server-ocis-storage.md
Comment (pascalwengerter) [inline tests/acceptance/expected-failures-with-ocis-server-ocis-storage.md:125]: Why are you overwriting the expected failure here instead of adding yours below?

File: tests/acceptance/stepDefinitions/filesContext.js
Comment (pascalwengerter) [inline tests/acceptance/stepDefinitions/filesContext.js:700]:
```suggestion
When('the user downloads a previous version of the file using the webUI', function () {
```

**Review by pascalwengerter — CHANGES_REQUESTED** [review]

File: tests/acceptance/expected-failures-with-ocis-server-ocis-storage.md
Comment (ScharfViktor) [reply tests/acceptance/expected-failures-with-ocis-server-ocis-storage.md:125]: The `https://github.com/owncloud/ocis/issues/2319` issue was closed due to the inability to download the file into the "Shares" folder, and the test for this case has been changed (the `versions.feature:77` test now passes).

File: tests/acceptance/stepDefinitions/filesContext.js
Comment (ScharfViktor) [reply tests/acceptance/stepDefinitions/filesContext.js:700]: fixed
@pascalwengerter thank you fo review))

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/20421/17/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/20426/67/1
The following scenarios passed on retry:
- webUIUpload/upload.feature:73

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles1</strong> https://drone.owncloud.com/owncloud/web/20426/56/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/20427/17/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by pascalwengerter** [comment]
@ScharfViktor could you investigate why the tests failed here? :)

**Comment by ScharfViktor** [comment]
> @ScharfViktor could you investigate why the tests failed here? :)

Strange things happen here)
When we create a new version in oC10, we get a version counter "0", when we create version 2, the version counter becomes "1". In ocis everything works correctly

**Comment by pascalwengerter** [comment]
> > @ScharfViktor could you investigate why the tests failed here? :)
> 
> Strange things happen here) When we create a new version in oC10, we get a version counter "0", when we create version 2, the version counter becomes "1". In ocis everything works correctly

I think if you search issues in oCIS/web you should find mentions for this. @individual-it and @phil-davis also should be able to provide more insights on this topic since it for sure already popped up in CI

**Comment by individual-it** [comment]
versions tests have often issues because in oC10 a version is only saved once per second. If a file is updated multiple times within one second it will only generate one version.
In web UI tests this was recently fixed by @SwikritiT  see https://github.com/owncloud/web/pull/6047

**Comment by individual-it** [comment]
the solution will be improved soon to be even more stable, see https://github.com/owncloud/web/issues/6081 and then it should be ported over to the middleware

**Comment by pascalwengerter** [comment]
@ScharfViktor could you soft reset your changes `git reset --soft HEAD~4` and squash them in one commit? There's some cyrillic letter in the commit messages and too many commits for such a small change anyways :)

**Comment by ScharfViktor** [comment]
> @ScharfViktor could you soft reset your changes `git reset --soft HEAD~4` and squash them in one commit? There's some cyrillic letter in the commit messages and too many commits for such a small change anyways :)

Yeah sure, I'll do it. I just have to wait until the CI is green.

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles2</strong> https://drone.owncloud.com/owncloud/web/20985/56/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by pascalwengerter** [comment]
> > @ScharfViktor could you soft reset your changes `git reset --soft HEAD~4` and squash them in one commit? There's some cyrillic letter in the commit messages and too many commits for such a small change anyways :)
> 
> Yeah sure, I'll do it. I just have to wait until the CI is green.

That doesn't make too much sense since it would need to run once again after you did it anyways, no?

**Comment by ScharfViktor** [comment]
The test `versions.feature:63` used to work in Osis, because when we created the new version - the number of versions was 0 (it was wrong). When all this was fixed, the test failed

steps `And the administrator has cleared the versions for all users` doesnot work in oCis. 

`Given('the administrator has cleared the versions for all users', function () {
  if (client.globals.ocis) {
    // TODO: decide if we fail on OCIS when a scenario even tries to use this given step
    return
  }
  return occHelper.runOcc(['versions:cleanup'])
})`

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingAccept</strong> https://drone.owncloud.com/owncloud/web/20990/15/1
The following scenarios passed on retry:
- webUISharingAcceptSharesToRoot/acceptShares.feature:108

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/20990/17/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: tests/acceptance/expected-failures-with-ocis-server-ocis-storage.md
Comment (pascalwengerter) [reply tests/acceptance/expected-failures-with-ocis-server-ocis-storage.md:119]: Failed with line 77 🤔 https://drone.owncloud.com/owncloud/web/20990/17/22

**Comment by ownclouders** [comment]
Results for <strong>oC10Files2</strong> https://drone.owncloud.com/owncloud/web/21013/17/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Comment by ownclouders** [comment]
Results for <strong>oCISFiles2</strong> https://drone.owncloud.com/owncloud/web/21071/56/1
:boom: The acceptance tests pipeline failed. The build has been cancelled.

**Review by pascalwengerter — APPROVED** [review]
Fine for me, not sure if we need to sort out expected failures after this has been merged

**Pull request merged**
